state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1269,87 +1359,108 @@ pub mod pallet {
-						ccm_id,
-						PendingCcms::<T>::take(ccm_id).expect("Ccm can only be completed once."),
-						(principal, gas),
-					);
-					*maybe_ccm_output = None;
-				}
+		fn schedule_swap(
+			input_asset: Asset,
+			output_asset: Asset,
+			input_amount: AssetAmount,
+			refund_params: Option<SwapRefundParameters>,
+			swap_type: SwapType,
+			swap_request_id: SwapRequestId,
+		) -> SwapId {
+			let swap_id = SwapIdCounter::<T>::mutate(|id| {
+				id.saturating_accrue(1);
+				*id
 			});
-		}
 
-		/// Schedule the egress of a completed Cross chain message.
-		fn schedule_ccm_egress(
-			ccm_id: u64,
-			ccm_swap: CcmSwap,
-			(ccm_output_principal, ccm_output_gas): (AssetAmount, AssetAmount),
-		) {
-			// Schedule the given ccm to be egressed and deposit a event.
-			if let Ok(ScheduledEgressDetails { egress_id, egress_amount, fee_withheld }) =
-				T::EgressHandler::schedule_egress(
-					ccm_swap.destination_asset,
-					ccm_output_principal,
-					ccm_swap.destination_address.clone(),
-					Some((ccm_swap.deposit_metadata, ccm_output_gas)),
-				) {
-				if let Some(swap_id) = ccm_swap.principal_swap_id {
-					Self::deposit_event(Event::<T>::SwapEgressScheduled {
-						swap_id,
-						egress_id,
-						asset: ccm_swap.destination_asset,
-						amount: egress_amount,
-						fee: fee_withheld,
-					});
-				}
-				Self::deposit_event(Event::<T>::CcmEgressScheduled { ccm_id, egress_id });
-			} else {
-				log_or_panic!("CCM egress scheduling should never fail.");
-			}
+			let execute_at = frame_system::Pallet::<T>::block_number() + SWAP_DELAY_BLOCKS.into();
+
+			SwapQueue::<T>::append(
+				execute_at,
+				Swap::new(
+					swap_id,
+					swap_request_id,
+					input_asset,
+					output_asset,
+					input_amount,
+					refund_params,
+					swap_type,
+				),
+			);
+
+			Self::deposit_event(Event::<T>::SwapScheduled {
+				swap_request_id,
+				swap_id,
+				input_amount,
+				swap_type,
+				execute_at,
+			});
+
+			swap_id
 		}
 	}
 
-	impl<T: Config> SwapDepositHandler for Pallet<T> {
+	impl<T: Config> SwapRequestHandler for Pallet<T> {
 		type AccountId = T::AccountId;
 
-		/// Callback function to kick off the swapping process after a successful deposit.
-		fn schedule_swap_from_channel(
-			deposit_address: ForeignChainAddress,
-			deposit_block_height: u64,
-			from: Asset,
-			to: Asset,
-			amount: AssetAmount,
-			destination_address: ForeignChainAddress,
-			broker_commission: Beneficiaries<Self::AccountId>,
+		fn init_swap_request(
+			input_asset: Asset,
+			input_amount: AssetAmount,
+			output_asset: Asset,
+			request_type: SwapRequestType,
+			broker_fees: Beneficiaries<Self::AccountId>,
 			refund_params: Option<ChannelRefundParameters>,
-			channel_id: ChannelId,
-		) -> SwapId {
-			// Permill maxes out at 100% so this is safe.
-			let fee: u128 = Permill::from_parts(
-				broker_commission.iter().fold(0, |acc, entry| acc + entry.bps) as u32 *
-					BASIS_POINTS_PER_MILLION,
-			) * amount;
+			origin: SwapOrigin,
```

Comment:
Same with origin I guess, it only applies to user swaps, and its only use seems to be to add to events. Could just be embedded in the SwapType.

Reply:
How do you mean exactly? Whether the deposit comes from the vault contract or a deposit channel seems orthogonal to whether it is "regular"/ccm.

> and its only use seems to be to add to events.

To me that's the more reason to keep them separate, since it won't be needed after the initial event is emitted.